app/experimenter/nimbus-ui/src/components/ChangeApprovalOperations/index.tsx

```diff
@@ -0,0 +1,153 @@
+/* istanbul ignore file until EXP-1055 & EXP-1062 done */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useCallback, useState } from "react";
+import Alert from "react-bootstrap/Alert";
+import { ReactComponent as Check } from "../../images/check.svg";
+import { humanDate } from "../../lib/dateUtils";
+import FormApproveOrReject from "./FormApproveOrReject";
+import FormRejectReason from "./FormRejectReason";
+import FormRemoteSettingsPending from "./FormRemoteSettingsPending";
+import { NimbusChangeLogType } from "./temp-types";
+
+export enum ChangeApprovalOperationsState {
+  None,
+  ApprovalPending,
+  ApproveOrReject,
+  RemoteSettingsPending,
+  Rejection,
+  Rejected,
+  RejectedByAnotherUser,
+  TimedOut,
+}
+
+export type ChangeApprovalOperationsProps = {
+  actionDescription: string;
+  isLoading: boolean;
+  canReview: boolean;
+  reviewRequestEvent?: NimbusChangeLogType;
+  approvalEvent?: NimbusChangeLogType;
+  rejectionEvent?: NimbusChangeLogType;
+  timeoutEvent?: NimbusChangeLogType;
+  rejectChange: (fields: { reason: string }) => void;
+  approveChange: () => void;
+  startRemoteSettingsApproval: () => void;
+};
+
+export const ChangeApprovalOperations: React.FC<
+  React.PropsWithChildren<ChangeApprovalOperationsProps>
+> = ({
+  actionDescription,
+  isLoading = false,
+  canReview = false,
+  reviewRequestEvent,
+  approvalEvent,
+  rejectionEvent,
+  timeoutEvent,
+  rejectChange,
+  approveChange,
+  startRemoteSettingsApproval,
+  children,
+}) => {
+  let defaultUIState = ChangeApprovalOperationsState.None;
+  if (rejectionEvent) {
+    defaultUIState = ChangeApprovalOperationsState.Rejected;
+  } else if (timeoutEvent) {
+    defaultUIState = canReview
+      ? ChangeApprovalOperationsState.ApproveOrReject
+      : ChangeApprovalOperationsState.ApprovalPending;
+  } else if (approvalEvent) {
+    defaultUIState = canReview
+      ? ChangeApprovalOperationsState.RemoteSettingsPending
+      : ChangeApprovalOperationsState.ApprovalPending;
+  } else if (reviewRequestEvent) {
+    defaultUIState = canReview
+      ? ChangeApprovalOperationsState.ApproveOrReject
+      : ChangeApprovalOperationsState.ApprovalPending;
+  }
+  const [uiState, setUIState] = useState<ChangeApprovalOperationsState>(
+    defaultUIState,
+  );
+  const resetDraftUIState = useCallback(() => setUIState(defaultUIState), [
+    defaultUIState,
+    setUIState,
+  ]);
+
+  switch (uiState) {
+    case ChangeApprovalOperationsState.ApprovalPending:
+      return (
+        <Alert
+          data-testid="submit-success"
+          variant="success"
+          className="bg-transparent text-success"
+        >
+          <p className="my-1" data-testid="in-review-label">
+            <Check className="align-top" /> All set! This experiment will{" "}
```

Comment:
> Also, IMO, these views maybe don't look quite right for an ending experiment (i.e. should be warnings? not "all set"?):

Yeah, I feel like "All set!" is nice feedback if the user _just_ went through the flow, but probably not worth the trouble to set that up. A warning feels more appropriate to me personally with copy like, "This experiment will end as soon as the request to end is approved.", but seems fine for now. We could use a quick design review next week like we did for the review/reject flow and similarly, make a follow up for tweaks👍